internal/provider/edgegw/vpn_ipsec_resource.go

```diff
@@ -0,0 +1,522 @@
+	resp.Diagnostics.Append(d...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Set state to fully populated data
+	resp.Diagnostics.Append(resp.State.Set(ctx, stateRefreshed)...)
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (r *vpnIPSecResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	state := &VPNIPSecModel{}
+
+	// Get current state
+	resp.Diagnostics.Append(req.State.Get(ctx, state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Init the resource
+	resp.Diagnostics.Append(r.Init(ctx, state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	/*
+		Implement the resource deletion here
+	*/
+	// Lock object EdgeGateway
+	vdcOrVDCGroup, err := r.edgegw.GetParent()
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving Edge Gateway parent", err.Error())
+		return
+	}
+	if vdcOrVDCGroup.IsVDCGroup() {
+		mutex.GlobalMutex.KvLock(ctx, vdcOrVDCGroup.GetID())
+		defer mutex.GlobalMutex.KvUnlock(ctx, vdcOrVDCGroup.GetID())
+	} else {
+		mutex.GlobalMutex.KvLock(ctx, r.edgegw.GetID())
+		defer mutex.GlobalMutex.KvUnlock(ctx, r.edgegw.GetID())
+	}
+
+	// Get VPN
+	ipSecVPNConfig, err := r.edgegw.NsxtEdgeGateway.GetIpSecVpnTunnelById(state.ID.Get())
+	if err != nil {
+		resp.Diagnostics.AddError("Error Retrieving IPsec VPN Tunnel: %s", err.Error())
+		return
+	}
+
+	// Delete VPN
+	err = ipSecVPNConfig.Delete()
+	if err != nil {
+		resp.Diagnostics.AddError("Error Deleting IPsec VPN Tunnel configuration: %s", err.Error())
+		return
+	}
+}
+
+func (r *vpnIPSecResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	var (
+		edgegwID, edgegwName string
+		d                    diag.Diagnostics
+		err                  error
+		vpnIPSec             *govcd.NsxtIpSecVpnTunnel
+	)
+
+	// Split req.ID with dot. ID format is EdgeGatewayIDOrName.VPNIPSecID
+	idParts := strings.Split(req.ID, ".")
+
+	if len(idParts) != 2 {
+		resp.Diagnostics.AddError("Invalid ID format", "ID format is EdgeGatewayIDOrName.VPNIPSecIDOrName. If Several Name are the same, please use ID instead")
+		return
+	}
+
+	// Get Org to retrieve EdgeGateway
+	r.org, d = org.Init(r.client)
+	if d.HasError() {
+		resp.Diagnostics.Append(d...)
+		return
+	}
+
+	// Get EdgeGW is ID or Name
+	if uuid.IsEdgeGateway(idParts[0]) {
+		edgegwID = idParts[0]
+	} else {
+		edgegwName = idParts[0]
+	}
+
+	r.edgegw, err = r.org.GetEdgeGateway(edgegw.BaseEdgeGW{
+		ID:   types.StringValue(edgegwID),
+		Name: types.StringValue(edgegwName),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to import VPN IPSec.", err.Error())
+		return
+	}
+
+	// Get VPN IPSec
+	vpnIPSec, err = r.edgegw.GetIpSecVpnTunnelByName(idParts[1])
+	if govcd.ContainsNotFound(err) {
+		vpnIPSec, err = r.edgegw.GetIpSecVpnTunnelById(idParts[1])
+	}
+
+	var listStr string
+	if err != nil && !govcd.ContainsNotFound(err) {
```

Comment:
```suggestion
	if err != nil {
```

This is already tested